### PR TITLE
Postgres, Email, true shadowsocks encryption, OUTLINE ssconf subscription.

### PR DESCRIPTION
**Pull request by LaitmanX**
**PostgreSQL Support:**
Added Psycopg to the requirements, enabling out-of-the-box support for PostgreSQL, thereby expanding our database compatibility and offering more robust data management capabilities.

**Username Email Support:**
Extended the regular expression logic to support email addresses as usernames. This is a crucial step toward facilitating future payment subscription functionalities, ensuring a seamless user experience, especially in scenarios where email-based identification and communication are preferred.

**Shadowsocks Encryption Protocols Fix:**
Addressed a limitation in the Shadowsocks methods where only the poly1305 encryption protocol was effectively utilized. Despite the visual availability of other protocols, they were not operational at the encryption level. This fix ensures that all listed encryption protocols, including aes-128-gcm and aes-256-gcm, are now fully functional, thereby enhancing the security and flexibility of our encryption offerings.

**Outline Subscription Support or [Dynamic Access Keys](https://www.reddit.com/r/outlinevpn/wiki/index/dynamic_access_keys/):**
Implemented support for the ssconf://SUBSCRIPTION_LINK/outline or https://SUBSCRIPTION_LINK/outline format, facilitating the utilization of the Outline app subscription method. This enhancement simplifies the integration process and expands our subscription management capabilities, providing a more user-friendly and versatile approach to handle subscription links for Shadowsocks protocol what is mostly commonly used on Android TV and in WIFI Routers Configs just to bypass some common things and BTW much more stable than XRAY apps at that moment, so this is a good alternative in some contries without GFW.

Basically, OUTLINE **_Dynamic_** Access Keys follows the original JSON config format of shadowsocks. 
The required fields are:

{
"server": "198.199.101.152",
"server_port": 8388,
"password": "u1rRWTssNv0p",
"method": "aes-256-cfb",
"remarks": "Example 1"
}

### CODE REVIEW

**Comment by LaitmanX** [comment]
> nickname? there is no nickname that's username, and it should not be an Email
> 
> isn't chacha20-poly1305 same as chacha20-ietf-poly1305 ?
> 
> outline? is xray-core protocols same as outline protocols?

isn't chacha20-poly1305 same as chacha20-ietf-poly1305 ?

app/utils/share.py
`                    address: str,
                    port: int,
                    password: str,
                    security='chacha20-ietf-poly1305'):
                    method: str):`


There was typo bug and poly1305 was warking ad default security='chacha20-ietf-poly1305'), so i fixed a bug.

> nickname? there is no nickname that's username, and it should not be an Email
> 
> isn't chacha20-poly1305 same as chacha20-ietf-poly1305 ?
> 
> outline? is xray-core protocols same as outline protocols?

Username vs Email:
Using email as a username is a common practice, especially in the future context of paid subscriptions, as it provides a convenient way for account recovery, management, and an out-of-the-box subscription payment capability. However, I am open to discussion and willing to consider alternative methods of adding email if it aligns with the standards and requirements of the project.

chacha20-poly1305 vs chacha20-ietf-poly1305:
In fact, chacha20-poly1305 and chacha20-ietf-poly1305 have some differences in specifications, but the main point that was discovered and corrected in this pull request is a bug that limited the use to only chacha20-ietf-poly1305. Now, with the fix, all protocols are supported, including aes-128-gcm and aes-256-gcm, which enhances the flexibility and security of our project.

Outline and xray-core protocols:
The protocols used in Outline App are based on standard Shadowsocks (ss://), while xray-core can support a variety of other protocols. However, the key point is that the improvements I proposed introduce support for the basic subscription file configuration format of Outline(ssconf://), simplifying integration and enhancing compatibility.
Basically, OUTLINE follows the original JSON config format of shadowsocks. The required fields are:

{
  "server": "198.199.101.152",
  "server_port": 8388,
  "password": "u1rRWTssNv0p",
  "method": "aes-256-cfb",
  "remarks": "Example 1"
}

**Comment by SaintShit** [comment]
all changes are excellent, thank you.

**Comment by M03ED** [comment]
Some of users report this bug after this merge

**Comment by SaintShit** [comment]
> Some of users report this bug after this merge ! 

fixed 55facaacaeafa3a05b9c7098e9a041c5790cfaaa
